src/main/java/de/zalando/aruha/nakadi/config/NakadiConfig.java

```diff
@@ -86,18 +79,36 @@ public ZooKeeperHolder zooKeeperHolder() {
     }
 
     @Bean
+    @Profile("kafka")
     public KafkaLocationManager getKafkaLocationManager() {
         return new KafkaLocationManager();
     }
 
     @Bean
+    @Profile("kafka")
     public KafkaFactory kafkaFactory() {
         return new KafkaFactory(getKafkaLocationManager());
     }
 
     @Bean
-    public KafkaRepositorySettings topicRepositorySettings() {
+    @Profile("kafka")
+    public KafkaRepositorySettings kafkaRepositorySettings() {
         return new KafkaRepositorySettings();
     }
 
+    @Bean
+    public TopicRepository topicRepository() {
+        if (environment.acceptsProfiles("kafka")) {
```

Comment:
For me it also looks a bit like premature optimization. Until we really have other brokers I wouldn't introduce that profile. Having a lot of profiles and configurations (even if they have a usable default) makes the application more complex and leads to questions. I'd keep it as simple as possible.

Following this path we also might introduce profiles for Postgres and Zookeeper backends as well and maybe other things. That leads to a big mess up.